modules/cloudwatch-logs/main.tf

```
@@ -23,10 +23,21 @@ module "kfh" {
 }
 
 resource "aws_cloudwatch_log_subscription_filter" "this" {
-  count           = length(var.cloudwatch_log_groups)
+  count           = !var.use_order_independent_filter_resource_naming ? length(var.cloudwatch_log_groups) : 0
+
```

I think we leave this resource block unchanged as the output block switches over the new flag.

```suggestion
  count           = length(var.cloudwatch_log_groups)
```

---

But then you’ll have duplicate resources. Both use `var.cloudwatch_log_groups` to dynamically generate the subscription filters.

The idea was to add a flag to switch between the two resource naming conventions. I didn’t think there was a way to do this without a separate resource block.

---

Ah, yeah - I understand now. You're right, you can ignore my suggestion here 👍🏻